source/MaterialXCore/Document.cpp

```diff
@@ -825,12 +872,168 @@ void Document::upgradeVersion()
 
         minorVersion = 37;
     }
+    if (majorVersion == desiredMajorVersion &&
+        minorVersion == desiredMinorVersion)
+    {
+        setVersionString(makeVersionString(majorVersion, minorVersion));
+        return;
+    }
+
+    if (majorVersion == 1 && minorVersion == 37)
+    {
+        // Convert material Elements to Nodes
+        convertMaterialsToNodes(true);
+        minorVersion = 38;
+    }
+
+    if (majorVersion >= MATERIALX_MAJOR_VERSION &&
+        minorVersion >= MATERIALX_MINOR_VERSION)
+    {
+        setVersionString(makeVersionString(majorVersion, minorVersion));
+    }
+}
 
-    if (majorVersion == MATERIALX_MAJOR_VERSION &&
-        minorVersion == MATERIALX_MINOR_VERSION)
+bool Document::convertMaterialsToNodes(bool replaceNodes)
```

Comment:
This method might be better as a module-local function in the anonymous namespace, like updateChildSubclass above, since it's not part of the Document class interface that clients or subclasses would need to know about.